nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/dbcp/hive/HiveConnectionPool.java

```diff
@@ -348,6 +365,7 @@ public void onConfigured(final ConfigurationContext context) throws Initializati
 
         if (validationQuery != null && !validationQuery.isEmpty()) {
             dataSource.setValidationQuery(validationQuery);
+            dataSource.setValidationQueryTimeout(validationQueryTimeout); // timeout in seconds
```

Comment:
`DBCPConnectionPool` has a `getDataSource()` method in order to facilitate testing with mocks/stubs/delegates. It would be nice to have a unit test there and here (if possible) to exercise the timeout logic. Perhaps we can create a BasicDataSource delegate that returns mocked connections, and when a certain validation query is called, we sleep for a time longer than the timeout. I'm not exactly sure what that would look like, just tossing out an idea :) Have you tested this on a real system and verified the timeout occurs in such a situation?